pkg/assetvalidator/multivalidator.go

```diff
@@ -35,6 +35,18 @@ func (v *MultiAttributedAssetValidator) Validate(ctx context.Context, attributed
 		}
 	}
 	if len(errorList) > 0 {
+		var failureList []error
+		for _, err := range errorList {
+			switch err.(type) {
+			case ValidationFailure:
+				failureList = append(failureList, err)
+			default:
+				continue
+			}
+		}
+		if len(failureList) > 0 {
```

Comment:
Permanent/Non-retryable errors should take precedence here, because if there is even one of them, retrying the request would be useless.

Task: The condition for returning a ValidationFailure instead of a MultiValidatorError should be `if len(failureList) == len(errorList)` instead.

Reply:
isn't that what we're doing? if there is any failure list, we should find

Reply:
this now does the right thing